### PR TITLE
Make sure compilation of a module also compiles its tests

### PR DESCRIPTION
**Pull request by JaroslavTulach**
I'd like to contribute to the quality of NetBeans codebase. One of the common problems of our own harness (in contrary to Maven) is that by default compilation compiles only code, but not tests. As such, it often happens that developers forget to update the tests when refactoring their code. Yes, such mistake would ultimately be caught by Travis job, but checking for it during regular build would catch it sooner and bring the development experience closer to what people using Maven or Gradle expect.

I've done some testing and following commands seem to build OK with my change:
```bash
incubator-netbeans$ for i in *core*; do echo $i; ant -f $i/build.xml netbeans; done
incubator-netbeans$ for i in java*; do echo $i; ant -f $i/build.xml netbeans; done
```
thus my assumption is that the change is relatively safe for NetBeans own development.

On the other hand, I changed `common.xml` (I know it is better not to touch that file), and that may influence people using the harness to build their own applications. As such I am requesting review from Jesse.

### CODE REVIEW

**Comment by jlahoda** [comment]
I think this is a fine idea. The travis build failed, though, seems(?) that nbjunit is not yet built when the openide.util.lookup's tests are being built. Not sure why exactly is that, needs investigation.

**Comment by neilcsmith-net** [comment]
> We should just drop support for the Ant-based build harness IMO.

Surely _that_ would be a rather drastic change that should involve some discussions with RCP devs on the lists! (PR itself sounds a good idea)

**Comment by JaroslavTulach** [comment]
Thanks for the review, Jesse & co. As correctly observed by Jan, the build fails due to cyclic references. I could probably fix them by moving some tests from `nbjunit` to `openide.util.lookup`, but then I got a better idea: let's fix two issues at once!

1. build should build code & tests
2. there should be `ant build` target in module projects

It is very unusual that the Ant harness module `build.xml` files don't have `build` target and one has to use `ant netbeans` instead. I can fix it as well as fix the build of tests:

1. let's add `<target name="build" depends="netbeans,test-build"/>` into `common.xml`
2. let's modify default targets of all modules in the `incubator-netbeans` repository to call `build`
2.a should there be any problem, people can change it back to `netbeans`
3. make sure build from the IDE calls `build` and not `netbeans`

This seems like more compatible and less drastic change which achieves the same thing: developers working on their own modules compile their code as well as tests.

I was silent for a while as I was unsure whether this proposal is good, but I still (even after two weeks) think it is... Shall I implement it?

File: apisupport.ant/src/org/netbeans/modules/apisupport/project/ui/ModuleActions.java [hunk 1]
```diff
@@ -121,14 +123,9 @@ public ModuleActions(NbModuleProject project) {
 
     public void refresh() {
         Set<String> supportedActionsSet = new HashSet<String>();
-        globalCommands.put(ActionProvider.COMMAND_BUILD, new String[] {"netbeans"}); // NOI18N
+        globalCommands.put(ActionProvider.COMMAND_BUILD, new String[] {"build"}); // NOI18N
         globalCommands.put(ActionProvider.COMMAND_CLEAN, new String[] {"clean"}); // NOI18N
-        //a gross hack to prevent 158481 and the like
```
Comment (JaroslavTulach) [inline apisupport.ant/src/org/netbeans/modules/apisupport/project/ui/ModuleActions.java]: I think @mkleint has to review this one!

File: apisupport.ant/src/org/netbeans/modules/apisupport/project/ui/ModuleActions.java [hunk 2]
```diff
@@ -582,11 +579,27 @@ void doRun() {
                     if (targetNames == null) {
                         throw new IllegalArgumentException(command);
                     }
+                    if (command.equals(ActionProvider.COMMAND_BUILD) || command.equals(ActionProvider.COMMAND_REBUILD)) {
```
Comment (JaroslavTulach) [inline apisupport.ant/src/org/netbeans/modules/apisupport/project/ui/ModuleActions.java]: This part could be reviewed by @tzezula, thanks.

File: apisupport.ant/src/org/netbeans/modules/apisupport/project/ui/ModuleActions.java (hunk 1 above)
Comment (mkleint) [reply apisupport.ant/src/org/netbeans/modules/apisupport/project/ui/ModuleActions.java]: lol, was it me annoyed by the continually broken test classes?

**Comment by JaroslavTulach** [comment]
Hooray, NetBeans harness now behaves almost like Maven! Btw.

>> We should just drop support for the Ant-based build harness IMO.
> Surely that would be a rather drastic 

Yes, the change of whole NetBeans build to use for example Maven would be so drastic that I hope nobody is willing to undertake it.

> PR itself sounds a good idea

Nobody, except Jesse. Rather don't tease him, as he is capable to do it. The only hope I have is that @jglick is busy with other tasks and will not find enough time either.

**Pull request merged**